### PR TITLE
Add a music player

### PR DESCRIPTION
**Pull request by magcius**
Add the ability to play some chill tunes in a loop while going through
the FBE. There's a minimal rampout fade, but it's not hooked up yet.

We aren't going to install the actual music asset yet, pending approval
our creative team. Thankfully, it's super simple to drop in once we do
approve it.

[endlessm/eos-shell#4765]

### CODE REVIEW

File: gnome-initial-setup/gis-driver.c [hunk 1]
```diff
@@ -85,6 +86,8 @@ struct _GisDriverPrivate {
   gchar *lang_override;
   gchar *default_timezone;
 
+  GisMusicPlayer *music_player;
```
Comment (cosimoc) [inline gnome-initial-setup/gis-driver.c]: Nothing ever frees this

File: gnome-initial-setup/gis-driver.c [hunk 2]
```diff
@@ -312,6 +316,7 @@ gis_driver_startup (GApplication *app)
 
   priv->main_window = gis_window_new (driver);
   priv->assistant = gis_window_get_assistant (GIS_WINDOW (priv->main_window));
+  priv->music_player = gis_music_player_new ("file://" PKGDATADIR "/fbe.ogg");
```
Comment (cosimoc) [inline gnome-initial-setup/gis-driver.c]: Do we have `fbe.ogg` as part of this PR? It's fine if we don't, as long as the music player gracefully fails.
Also I wonder if we should not put the music into a GResource like we do for everything else

Comment (magcius) [inline gnome-initial-setup/gis-driver.c]: It is not, pending approval from Dalio. I can put in the temporary asset for now.

Comment (cosimoc) [inline gnome-initial-setup/gis-driver.c]: I think it's fine to leave the asset out for now, as long as it doesn't crash if the asset is not there

File: gnome-initial-setup/gis-music-player.c [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+G_DEFINE_TYPE_WITH_PRIVATE (GisMusicPlayer, gis_music_player, G_TYPE_OBJECT);
+
+static void
+gis_music_player_dispose (GObject *object)
+{
+  GisMusicPlayer *player = GIS_MUSIC_PLAYER (object);
+  GisMusicPlayerPrivate *priv = gis_music_player_get_instance_private (player);
+
+  g_clear_object (&priv->bus);
+  g_clear_object (&priv->playbin);
```
Comment (cosimoc) [inline gnome-initial-setup/gis-music-player.c]: Should also remove the `fade_timeout_id`
Also should chain up to parent

File: gnome-initial-setup/gis-music-player.c [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+      message->src == GST_OBJECT (priv->playbin))
+    {
+      gst_element_seek_simple (priv->playbin, GST_FORMAT_TIME,
+                               GST_SEEK_FLAG_SEGMENT,
+                               priv->loop_point);
+    }
+}
+
+GisMusicPlayer *
+gis_music_player_new (const char *uri)
```
Comment (cosimoc) [inline gnome-initial-setup/gis-music-player.c]: The uri should be a property of the object...
So this code should probably be moved either in the property setter or in one of the initialization functions

File: gnome-initial-setup/gis-music-player.c [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+  return player;
+}
+
+void
+gis_music_player_play (GisMusicPlayer *player)
+{
+  GisMusicPlayerPrivate *priv = gis_music_player_get_instance_private (player);
+  priv->playing = TRUE;
+  gst_element_set_state (priv->playbin, GST_STATE_PAUSED);
```
Comment (cosimoc) [inline gnome-initial-setup/gis-music-player.c]: Shouldn't this be `GST_STATE_PLAYING`?

Comment (cosimoc) [inline gnome-initial-setup/gis-music-player.c]: Also should probably cancel the fadeout if this is called again

File: gnome-initial-setup/gis-music-player.c [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    {
+      g_object_set (priv->playbin, "volume", 0.0, NULL);
+      priv->fade_timeout_id = 0;
+      gst_element_set_state (priv->playbin, GST_STATE_PAUSED);
+      return FALSE;
+    }
+}
+
+void
+gis_music_player_rampout (GisMusicPlayer *player)
```
Comment (cosimoc) [inline gnome-initial-setup/gis-music-player.c]: What happens when this is called and the rampout is already going on?

File: gnome-initial-setup/gis-music-player.c (hunk 5 above)
Comment (magcius) [inline gnome-initial-setup/gis-music-player.c]: No, it should be `PAUSED`. See the `on_state_changed` which sets it to PLAYING after the preroll is done. We need to hold off in order to do a proper segment seek.

File: gnome-initial-setup/gis-music-player.c (hunk 4 above)
Comment (magcius) [inline gnome-initial-setup/gis-music-player.c]: yeah but properties are stupidly hard to write and i hate them. give me gproperty

File: gnome-initial-setup/gis-music-player.c (hunk 5 above)
Comment (cosimoc) [inline gnome-initial-setup/gis-music-player.c]: Actually scrap my comment about `PLAYING. I see what you do in the callback

**Comment by wjt** [comment]
I don't think we're doing this (though we are playing a startup sound on some images: #268 ).

**Pull request closed without merging**